cpp/src/Slice/Parser.cpp

```diff
@@ -881,38 +884,36 @@ Slice::Contained::includeLevel() const
 bool
 Slice::Contained::hasMetadata(const string& directive) const
 {
-    return find(_metadata.begin(), _metadata.end(), directive) != _metadata.end();
+    return findMetadata(directive).has_value();
 }
 
-bool
-Slice::Contained::findMetadata(const string& prefix, string& output) const
+optional<string>
+Slice::Contained::findMetadata(const string& prefix) const
 {
     for (const auto& p : _metadata)
     {
         if (p.find(prefix) == 0)
         {
-            output = p;
-            return true;
+            return p;
         }
     }
-
-    return false;
+    return nullopt;
 }
 
-list<string>
```

Comment:
We have the alias, might as well use it.
And this makes it consistent with the other metadata functions (where we use the alias).